source/appModules/calculator.py

```diff
@@ -125,3 +141,12 @@ def script_calculatorResult(self, gesture):
 				if isinstance(resultsScreen, UIA) and resultsScreen.UIAElement.cachedClassName == "LandmarkTarget":
 					# And no, do not allow focus to move.
 					queueHandler.queueFunction(queueHandler.eventQueue, ui.message, resultsScreen.firstChild.name)
+
+	# Handle both number row and numpad with num lock on.
+	@scriptHandler.script(
+		gestures=[f"kb:{i}" for i in range(10)]
```

Comment:
Unfortunately this does not work on all keyboard localization. On French keyboard, the digits on the number row (on the alpha-numeric keyboard) are accessible in combination with the shift key.
For example, if you press the key above the 'E' (a bit on the right) you get the open parenthesis character. If you press shift in compination with the key above the 'E', you get the digit '4'.

Also to be noted: some characters on the number row are commands of the calculator (parenthesis, minus sign).
The characters that are on the keys 1 to 0 on the number row of the French keyboard are the following:
`&é"'(-è_çà`